### ISSUE TITLE
Unresponsive application causes partial system freeze

### ISSUE THREAD

**Issue by Octogonapus** [pop-os/pop#347]
**Distribution (run `cat /etc/os-release`):**
```
NAME="Pop!_OS"
VERSION="18.04 LTS"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Pop!_OS 18.04 LTS"
VERSION_ID="18.04"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
This does not relate to a specific application AFAIK.

**Issue/Bug Description:**
When an application becomes unresponsive, mouse clicks/scrolling become unresponsive. I can still move the mouse around, but I cannot click or scroll on any applications. Additionally, the super key is also unresponsive in opening the Activities view (i.e. the super key does nothing during this time).

The application I am referring to is a JVM application I work on, and it becomes unresponsive when it hits a breakpoint in IntelliJ. During this time, although the mouse and super key do not work, I can still alt-tab into IntelliJ and use the keyboard to force quit the application to regain control of my system.

**Steps to reproduce (if you know):**
Described above.

**Expected behavior:**
Unresponsive applications should not freeze the system in this way.

**Other Notes:**
My sample size here is just 1. If someone knows a way that I can make any arbitrary application seem unresponsive, then I will test this out with more applications (googling for this just brings up results of people troubleshooting unresponsive applications, unhelpful to me).

**Comment by brs17** [comment]
Not being able to reach the Activities view sounds like Gnome is crashing/freezing. After this happens, I take it you end up rebooting? After rebooting can you provide the output of `journalctl -b -1`? This will show us the logs from the previous boot. Hopefully/potentially we can figure out the culprit from there.

**Comment by Octogonapus** [comment]
No, I do not reboot. "I can still alt-tab into IntelliJ and use the keyboard to force quit the application to regain control of my system." Are there some other logs I can look into instead?

**Comment by brs17** [comment]
Yes, just go ahead and provide the output of `journalctl`. There may be quite a lot of information so providing that as an attachment would be much appreciated.

Sounding to me like an issue with Java/JVM, but we'll see!

**Comment by Octogonapus** [comment]
Here it is: https://pastebin.com/LdstYMAY
I started viewing the log just before I hit the breakpoint. You can see `pushModal: invocation of begin_modal failed` every time I hit the super key, but I can't see anything else from the log. What do you think?

**Comment by brs17** [comment]
Have you tried watching your system (CPU/memory) performance when you reach this breakpoint? Rereading your issue, it almost sounds to me that if this keeps happening at that breakpoint, it might be bad code :stuck_out_tongue:

**Comment by Octogonapus** [comment]
Yes, I don't notice any performance anomalies. It's starting to seem to me like this is an IntelliJ/JVM issue. Some breakpoints don't cause any problems. This breakpoint in particular lives inside of a change listener.

**Comment by brs17** [comment]
I think I'm in agreement with you there that it seems to be more of an issue with IntelliJ. Perhaps try recreating the same issue on Ubuntu?

**Comment by Octogonapus** [comment]
I have reproduced this on Ubuntu. A colleague of mine, @madhephaestus, gave me the idea (because he encountered a very similar problem in the past) that the problem could be because my breakpoint is in the UI thread. When I hit th breakpoint, some UI elements (like the super key thing) become unresponsive.

I am able to use my keyboard to navigate back to IntelliJ and resume program execution, at which point GNOME becomes fully responsive again (the super key works again, for example). I now believe that this is a GNOME problem, what do you think? It seems to me that the application should be "sanboxed" in a sense. If my application's UI thread freezes, that shouldn't also cause GNOME to freeze.

**Comment by Octogonapus** [comment]
I am willing to put the time in to investigate and help fix this, but I need some sort of a lead to go on. If this might be a GNOME problem is there a utility I can use or a place I can look to search deeper into GNOME?
